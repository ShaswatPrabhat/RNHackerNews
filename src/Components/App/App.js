/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, { useEffect, useState } from "react";
import type { Node } from "react";
import AppWrapper from "../AppWrapper/AppWrapper";
import { Linking, Text } from "react-native";
import axios from "axios";
import { Button } from "@rneui/themed";

const App: () => Node = () => {
  const [data, setData] = useState(null);
  useEffect(() => {
    const fetchData = async () => {
      const response = await axios.get("https://news.ycombinator.com/news");
      setData(response);
    };
    fetchData();
  }, []);

  return (
    <AppWrapper>
      <Button
        type="outline"
        title="Bubu?"
        onPress={async () => {
          await Linking.openURL("https://news.ycombinator.com/news");
        }}
      />
      <Text>{JSON.stringify(data)}</Text>
    </AppWrapper>
  );
};

export default App;
