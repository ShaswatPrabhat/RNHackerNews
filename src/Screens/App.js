/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from "react";
import type { Node } from "react";
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  useColorScheme,
  View,
} from "react-native";

import { Colors, Header } from "react-native/Libraries/NewAppScreen";

const App: () => Node = () => {
  const isDarkMode = useColorScheme() === "dark";

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? "light-content" : "dark-content"} />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}
      >
        <Header />
        <View
          style={{
            backgroundColor: isDarkMode ? Colors.black : Colors.white,
            flex: 1,
          }}
        ></View>
      </ScrollView>
    </SafeAreaView>
  );
};

export default App;
