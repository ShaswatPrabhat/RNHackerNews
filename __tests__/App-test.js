/**
 * @format
 */

import "react-native";
import React from "react";
import App from "../src/Screens/App";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

it("renders correctly", () => {
  expect(renderer.create(<App />)).toMatchSnapshot();
});
