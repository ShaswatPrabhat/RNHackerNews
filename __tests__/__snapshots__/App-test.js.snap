// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      Object {
        "backgroundColor": "#F3F3F3",
      }
    }
  >
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Array [
            Object {
              "paddingBottom": 40,
              "paddingHorizontal": 32,
              "paddingTop": 96,
            },
            Object {
              "backgroundColor": "#F3F3F3",
            },
          ]
        }
      >
        <Image
          accessibilityRole="image"
          source={
            Object {
              "testUri": "../Libraries/NewAppScreen/components/logo.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "marginBottom": -192,
                "marginLeft": -128,
                "opacity": 0.2,
                "overflow": "visible",
                "resizeMode": "cover",
              },
            ]
          }
          testID="new-app-screen-header"
        />
        <Text
          style={
            Array [
              Object {
                "fontSize": 40,
                "fontWeight": "700",
                "textAlign": "center",
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Welcome to
          

          React Native
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
            "flex": 1,
          }
        }
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
